crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -196,6 +194,21 @@ impl TimelineState {
         txn.commit();
     }
 
+    fn should_add_event_item(
+        timeline_focus: &TimelineFocusKind,
+        thread_root: &Option<OwnedEventId>,
+    ) -> bool {
+        match &timeline_focus {
+            TimelineFocusKind::Live => thread_root.is_none(),
+
+            TimelineFocusKind::Thread { root_event_id } => {
+                thread_root.as_ref().is_some_and(|r| r == root_event_id)
+            }
+
+            TimelineFocusKind::Event | TimelineFocusKind::PinnedEvents => false,
+        }
+    }
```

Comment:
I'm a bit worried that we have an undocumented function, that partially replicates some behavior of `Transaction::should_add_event_item`; let's see if we can share more code here.